tests/Endpoints/MultiModalSearchTest.php

```diff
@@ -0,0 +1,216 @@
+                                'text' => '{{media.textAndPoster.text}}',
+                            ],
+                            [
+                                'type' => 'image_base64',
+                                'image_base64' => 'data:{{media.textAndPoster.image.mime}};base64,{{media.textAndPoster.image.data}}',
+                            ],
+                        ],
+                    ],
+                ],
+                'text' => [
+                    'value' => [
+                        'content' => [
+                            [
+                                'type' => 'text',
+                                'text' => '{{media.text.text}}',
+                            ],
+                        ],
+                    ],
+                ],
+                'poster' => [
+                    'value' => [
+                        'content' => [
+                            [
+                                'type' => 'image_url',
+                                'image_url' => '{{media.poster.poster}}',
+                            ],
+                        ],
+                    ],
+                ],
+            ],
+            'request' => [
+                // This request object matches the Voyage API request object
+                'inputs' => ['{{fragment}}', '{{..}}'],
+                'model' => 'voyage-multimodal-3',
+            ],
+            'response' => [
+                // This response object matches the Voyage API response object
+                'data' => [
+                    [
+                        'embedding' => '{{embedding}}',
+                    ],
+                    '{{..}}',
+                ],
+            ],
+        ];
+    }
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $http = new Client($this->host, getenv('MEILISEARCH_API_KEY'));
+        $http->patch('/experimental-features', ['multimodal' => true]);
+
+        $apiKey = getenv('VOYAGE_API_KEY');
+        if (false === $apiKey || '' === $apiKey) {
+            $this->voyageApiKey = null;
+
+            return; // This test case is skipped if the Voyage API key is not set
+        } else {
+            $this->voyageApiKey = $apiKey;
+        }
+
+        $this->index = $this->createEmptyIndex($this->safeIndexName());
+        $updateSettingsPromise = $this->index->updateSettings([
+            'searchableAttributes' => ['title', 'overview'],
+            'embedders' => [
+                'multimodal' => $this->getEmbedderConfig($this->voyageApiKey),
+            ],
+        ]);
+        $this->index->waitForTask($updateSettingsPromise['taskUid']);
+
+        // Load the movies.json dataset
+        $fileJson = fopen('./tests/datasets/movies.json', 'r');
+        $documentJson = fread($fileJson, filesize('./tests/datasets/movies.json'));
+        fclose($fileJson);
+        $this->documents = json_decode($documentJson, true);
```

Comment:
you access this only in a single test, maybe it should be moved to the test itself?

Reply:
Hmm, I missed that for some reason, lgtm then